.github/workflows/update-sboms.yaml

```diff
@@ -25,7 +25,7 @@ jobs:
       with:
         python-version: ${{ env.PYTHON_VERSION }}
 
-    - uses: actions/cache@v2
+    - uses: actions/cache@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
```

Comment:
Why are we using actions/cache at 4.0.0 here, but at 4.2.0 below in `.github/workflows/validations.yaml`? Should this line be changed to use 4.2.0?